### PR TITLE
feat: Add room title and editing functionality

### PR DESCRIPTION
**Pull request by pilanop**
In this pull request, I have added the feature to fetch and edit the title of a room in our application. Here's a brief summary of the changes:

- **Editing Room Title:** Users can now edit the title of the room from the frontend. The new title is sent to the backend and updated in the database.

## 🔄 Changes to Existing Code:

- Updated the room component to include the new title input field and its functionality.
- Added a new endpoint in the API to handle updating the room title.
- Updated the room schema in the database to include the title.

## 🔮 Future Plans:

- **Encrypting Title:** In future iterations, planed to enhance security by encrypting the room title before storing it in the database.

also fixed the issue [#13)

### CODE REVIEW

**Comment by RobiMez** [comment]
Reviewinn

File: src/routes/api/prof/+server.ts [hunk 1]
```diff
@@ -1,22 +1,17 @@
...
+  const { pbKey } = body;
   const profanityEnabledStatus = body.profanityEnabled;
   console.log('PATCH /api/prof/:pbKey called with body:', body);
   try {
     let room;
 
     if (profanityEnabledStatus === undefined) {
       // If profanityEnabled is not in the body, find the room and return its state
-      room = await Listener.findOne({ rid: rid }, { profanityEnabled: 1, _id: 0 });
+      room = await Listener.findOne({ pbKey: pbKey }, { profanityEnabled: 1, _id: 0 });
```
Comment (RobiMez) [inline src/routes/api/prof/+server.ts:14]: ooh yeah we did this cus @doniverse was having issues with public key based queries

File: src/routes/api/titl/+server.ts
Comment (RobiMez) [inline src/routes/api/titl/+server.ts:1]: LOL love the folder naming 😆

File: src/routes/api/prof/+server.ts (hunk 1 above)
Comment (pilanop) [reply src/routes/api/prof/+server.ts:14]: Sure sure u can change it back

Comment (RobiMez) [reply src/routes/api/prof/+server.ts:14]: noice , btw love what you did. im looking thru the code making some svelte reactivity changes. first load was giving me undefined. 

also updated the prod database with the title field 
image

**Comment by RobiMez** [comment]
ALL GOOD !

**Pull request merged**